server/route.go

```diff
@@ -1388,17 +1388,22 @@ func (s *Server) updateRouteSubscriptionMap(acc *Account, sub *subscription, del
 	// here but not necessarily all updates need to be sent. We need to block and recheck the
 	// n count with the lock held through sending here. We will suppress duplicate sends of same qw.
 	if isq {
+		// We can't keep the account's lock while getting a connection's lock,
+		// so we use another lock to protect the send for this account.
+		acc.sendQMu.Lock()
+		defer acc.sendQMu.Unlock()
```

Comment:
Yes because the things that are changed below need to be protected (since other threads may end in processSub/Unsub and lock the account to do the subject accounting).

Reply:
The other approach would be to, if `isq`, lock sqmu and defer unlock at the top of the function and then we don't need lqws (and therefore the second check). wdyt?

Reply:
I think that might be better.

Reply:
I think so too.. running tests as I type.

Reply:
I have made the changes and force-pushed it..